pisa/stages/osc/layers.py

```diff
@@ -101,14 +101,11 @@ def extCalcLayers(
         traverse_dist = np.zeros(max_layers, dtype=FTYPE)
         traverse_electron_frac = np.zeros(max_layers, dtype=FTYPE)
 
+        r_prop = self.r_detector + self.detector_depth + self.prop_height
+        path_len = -self.r_detector*coszen + np.sqrt(self.r_detector**2.*coszen**2 - (self.r_detector**2. - r_prop*r_prop))
+
         # Above horizon
         if coszen >= 0:
-            kappa = (detector_depth + prop_height)/r_detector
-            path_len = (
-                r_detector * np.sqrt(coszen**2 - 1 + (1 + kappa)**2)
-                - r_detector * coszen
-            )
-
             # Path through the air:
             kappa = detector_depth / r_detector
```

Comment:
This line was particularly dangerous, as it redefined the variable kappa mid- condition. Taking the total path length out of this loop gets rid of this issue